### ISSUE TITLE
Ibus is not working in chromium browser!

### ISSUE THREAD

**Issue by Mahmudul-hasan613** [ibus/ibus#2577]
Which distribution and version?: 
Ubuntu 23.10

Which desktop environment and version?: 
GNOME version 45.0

Which session type?: 
X11

Which application and version?:
Chromium (version : 118.0.5993.117 (official build) snap (64-bit))

IBus version?: 
IBus 1.5.29-rc1

Issue description:

Can you reproduce your problem when you restart ibus-daemon? (yes / no): 
Yes.

Do you see any errors when you run ibus-daemon with the verbose option?:
I'm not sure! Screenshot is attached.
![2023-10-31 00-41-04 হতে স্ক্রিনশট](https://github.com/ibus/ibus/assets/60663471/fe699035-4328-4013-bf31-4e0b159164c0)

Can you reproduce your problem with a new user account instead of the current your account? (yes / no): 
I'm not sure. But I think the problem can be reproduced.

**Comment by fujiwarat** [comment]
I guess you would miss some settings.

> I'm not sure! Screenshot is attached.

You have to terminate ibus-daemon before you run it.
```
% ibus exit
% ibus-daemon --xim --verbose &
```

Is your problem Chromium only?

What is the output of `ibus im-module --type gtk3` ?

**Comment by Mahmudul-hasan613** [comment]
> You have to terminate ibus-daemon before you run it.

The ibus exit command causes the keyboard to stop working. After logging out and logging in again, it works.

> Is your problem Chromium only?

Yes.

> What is the output of `ibus im-module --type gtk3` ?

ibus

**Comment by fujiwarat** [comment]
> The ibus exit command causes the keyboard to stop working. After logging out and logging in again, it works.

Then you need to run `ibus exit` on an application not to enable ibus likes `env XMODIFIERS=@im=none xterm`

**Comment by Mahmudul-hasan613** [comment]
After following the procedure you mentioned, the following screenshot shows the result.
![2023-11-06 22-03-46 হতে স্ক্রিনশট](https://github.com/ibus/ibus/assets/60663471/d66c325b-7389-4723-8a28-085470caae05)

**Comment by fujiwarat** [comment]
Can you check any warning messages with `verbose` option besides `xim` option?

**Comment by Mahmudul-hasan613** [comment]
> Can you check any warning messages with `verbose` option besides `xim` option?

No.

**Comment by fujiwarat** [comment]
Can you check any warning messages with verbose option besides xim option?

    No.

Why cannot you do it? I think you should try it.

**Comment by Mahmudul-hasan613** [comment]
> Why cannot you do it? I think you should try it.

Because I do not know how to check the warning message!

**Comment by Mahmudul-hasan613** [comment]
> ```
>     Can you check any warning messages with verbose option besides xim option?
> 
> No.
> ```
> 
> Why cannot you do it? I think you should try it.

Because I do not know how to check the warning message! Can you please teach me how to check it's warning message?

**Comment by fujiwarat** [comment]
> Because I do not know how to check the warning message! Can you please teach me how to check it's warning message?

You can run `ibus-daemon --xim --verbose` and check if that command or  chromium command outputs any warnings when you run chromium browser.

**Comment by Mahmudul-hasan613** [comment]
> > Because I do not know how to check the warning message! Can you please teach me how to check it's warning message?
> 
> You can run `ibus-daemon --xim --verbose` and check if that command or chromium command outputs any warnings when you run chromium browser.

The screenshot shows the result after inputting the command you mentioned.
![2023-11-11 23-12-27 হতে স্ক্রিনশট](https://github.com/ibus/ibus/assets/60663471/f67f24b8-4d27-4d15-9bcd-216a822ec2c6)

Chromium browser is launched via terminal command. Then enter the command mentioned above; Then the message that appears in the terminal is shown in the screenshot.
![2023-11-11 23-14-02 হতে স্ক্রিনশট](https://github.com/ibus/ibus/assets/60663471/6289a8c8-3ae2-475d-8e7a-e9007817d1e4)

**Comment by mike-fabian** [comment]
I wonder whether that has anything to with that chromium is installed using snap on Ubuntu. In my Ubuntu 23.10 virtual machine I have:

```
mike-fabian@ubuntu23:~$ snap info chromium
name:      chromium
summary:   Chromium web browser, open-source version of Chrome
publisher: Canonical✓
store-url: https://snapcraft.io/chromium
contact:   https://bugs.launchpad.net/ubuntu/+source/chromium-browser/+bugs?field.tag=snap
license:   unset
description: |
  An open-source browser project that aims to build a safer, faster, and more
  stable way for all Internet users to experience the web.
commands:
  - chromium.chromedriver
  - chromium
snap-id:      XKEcBqPM06H1Z7zGOdG5fbICuf8NWK5R
tracking:     latest/stable
refresh-date: yesterday at 22:17 CET
channels:
  latest/stable:    119.0.6045.123 2023-11-09 (2686) 165MB -
  latest/candidate: 119.0.6045.123 2023-11-08 (2686) 165MB -
  latest/beta:      120.0.6099.18  2023-11-09 (2689) 166MB -
  latest/edge:      121.0.6115.2   2023-11-10 (2691) 167MB -
installed:          119.0.6045.123            (2686) 165MB -
mike-fabian@ubuntu23:~$ cat /etc/os-release
PRETTY_NAME="Ubuntu 23.10"
NAME="Ubuntu"
VERSION_ID="23.10"
VERSION="23.10 (Mantic Minotaur)"
VERSION_CODENAME=mantic
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=mantic
LOGO=ubuntu-logo
mike-fabian@ubuntu23:~$

```

Nevertheless, ibus works for me when using Gnome Wayland on that machine, see this screenshot:

![Screenshot](https://github.com/ibus/ibus/assets/2330175/7188dad9-722c-44db-94d1-20b4712ae4b2)

I cannot try Xorg on that Virtual machine as the X server crashes immediately when I try to log into Gnome Xorg.

**Comment by fujiwarat** [comment]
I think chromium-browser uses GTK3 and what is the result of running a command `ibus im-module --type gtk3` ?

**Comment by Mahmudul-hasan613** [comment]
> I think chromium-browser uses GTK3 and what is the result of running a command `ibus im-module --type gtk3` ?

ibus

![2023-11-14 23-35-05 হতে স্ক্রিনশট](https://github.com/ibus/ibus/assets/60663471/f3c3c009-6b17-4946-a80d-39be1c52fe69)

**Comment by fujiwarat** [comment]
I don't find any problems from your replies at the moment.

> Can you reproduce your problem with a new user account instead of the current your account? (yes / no):
> I'm not sure. But I think the problem can be reproduced.

Can you try this? you can create a new "tester" account with the following steps and log out the current session.
```
% sudo useradd -d /home/tester -m tester
% sudo passwd tester
```

Which ibus engine do you use? You can check the current engine name with `ibus engine` command.

Can you attach the output of `strace /snap/chromium/current/usr/lib/chromium-browser/chrome |& grep ibus` after you install strace?

**Comment by Mahmudul-hasan613** [comment]
> I don't find any problems from your replies at the moment.
> 
> > Can you reproduce your problem with a new user account instead of the current your account? (yes / no):
> > I'm not sure. But I think the problem can be reproduced.
> 
> Can you try this? you can create a new "tester" account with the following steps and log out the current session.
> 
> ```
> % sudo useradd -d /home/tester -m tester
> % sudo passwd tester
> ```
ibus working in Chromium on tester user.
> Which ibus engine do you use? You can check the current engine name with `ibus engine` command.
![2023-11-16 00-00-54 হতে স্ক্রিনশট](https://github.com/ibus/ibus/assets/60663471/49e939a6-d1b8-43fc-b801-97c0a1a036ee)
> Can you attach the output of `strace /snap/chromium/current/usr/lib/chromium-browser/chrome |& grep ibus` after you install strace?
![2023-11-15 23-58-39 হতে স্ক্রিনশট](https://github.com/ibus/ibus/assets/60663471/4bb8b45d-5784-45d2-ab0e-2b429c502a13)


**Please tell me how to delete the tester user?**

**Comment by fujiwarat** [comment]
> ibus working in Chromium on tester user.

It means your original account has a configuration error. It's good to check for you what is different between the original account and 'tester' user.
Regarding to the  ibus configuration, you could read the result of `ibus read-config` and `ibus read-cache` commands.
Also you may like to check `gnome-control-center` settings.

> ibus is working when I launch chrome browser using `strace /snap/chromium/current/usr/lib/chromium-browser/chrome |& grep ibus` command.

That's strange. You have a different configuration between `/usr/bin/chromium-browser` and `/snap/chromium/current/usr/lib/chromium-browser/chrome` .
It would be good for you to investigate the difference by yourself or you may like to ask your downstream Ubuntu community.
You can check the chromium-browser behavior with `sh -x /usr/bin/chromium-browser` but I'm not familiar with snap.

> Please tell me how to delete the tester user?
`sudo userdel tester` and you may like to delete /home/tester

**Comment by Mahmudul-hasan613** [comment]
My problem is solved. But how was it resolved; I'm not sure about that! First of all it asked me for admin password because of using strace command. Giving that might have fixed my problem. Or there may be another reason. That is: First I went to Resion & Language from gnome control settings, from there I went to Manage Installed Languages. After going to that option some ibus related files are automatically downloaded. Then select ibus instead of none in Keyboard Input Method System. Now I can write Bengali using ibus in chromium browser.
Thank you very much for your outstanding support. I will be forever indebted to you for this.
